agent/lib/kontena/launchers/etcd.rb

```diff
@@ -85,20 +85,31 @@ def create_data_container(image)
     # @param [String] image
     # @param [Hash] info
     def create_container(image, info)
+      cluster_size = info['grid']['initial_size']
+      node_number = info['node_number']
+      cluster_state = 'new'
+      weave_ip = weave_ip(info['node_number'])
+
       container = Docker::Container.get('kontena-etcd') rescue nil
       if container && container.info['Config']['Image'] != image
         container.delete(force: true)
       elsif container && container.running?
-        info "etcd is already running"
+        info 'etcd is already running'
         @running = true
-        return
+        return container
+      elsif container && !container.running?
+        info 'etcd container exists but not running, starting it'
+        container.start
+        @running = true
```

Comment:
Are we using this `@running` information anywhere? If not, maybe we should remove `@running` and `#running?`.